app/download/views.py

```diff
@@ -120,6 +121,19 @@ def check_scan_verdict(service_id, document_id):
         )
         return jsonify(error=str(e)), MALICIOUS_CONTENT_ERROR_CODE
     except ScanInProgressError as e:
+        age_seconds = scan_files_document_store.get_object_age_seconds(
+            service_id, document_id, sending_method
+        )
+        if age_seconds > SCAN_TIMEOUT_SECONDS:
+            current_app.logger.info(
+                "Scan timed out for document: {}".format(e),
+                extra={
+                    "service_id": service_id,
+                    "document_id": document_id,
+                },
+            )
+            return jsonify(scan_verdict="scan_timed_out"), 200
```

Comment:
Will that be a `200` http status code if the scan timed out? Do we want to send something more specific to send to the API?

Reply:
I chose `200` because I wanted this to work with the existing logic in notification-api and to result in a send, so we would need to change the logic over there if we return a different error code: https://github.com/cds-snc/notification-api/blob/main/app/delivery/send_to_providers.py#L158

The server will send the json data `{"scan_verdict": "scan_timed_out"}` in the response so I figured we could read that if we want to detect this in notification-api in the future.

Reply:
The status code would be an option more aligned with HTTP (or RESTful), as no parsing would be required. There is even a status code dedicated for it, i.e. [408](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/408).

Reply:
`200` usually always means a success, but it's not a success here, so that could be misleading, albeit only to developers of Notify as this is an internal call.

Reply:
Also (sorry to add more), and that could be more serious, if some library interprets this code, they will assume it's a success. We might not be affected now but we could in the future (API calls reporting success would be all `200` for example -- but that should be an error).

Reply:
Fair enough, I will modify the status code in 2 follow up PRs, one on the dd-api and one on notification-api.